pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2606,40 +2606,30 @@ private void createPendingLoadTopic() {
-                                pulsar.getExecutor().execute(() -> {
-                                    // If topic is already exist, creating partitioned topic is not allowed.
-
-                                    if (metadata.partitions == 0
-                                            && !topicExists
-                                            && !topicName.isPartitioned()
-                                            && pulsar.getBrokerService().isAllowAutoTopicCreation(topicName)
-                                            && pulsar.getBrokerService().isDefaultTopicTypePartitioned(topicName)) {
-
-                                        pulsar.getBrokerService().createDefaultPartitionedTopicAsync(topicName)
-                                                .thenAccept(md -> future.complete(md))
-                                                .exceptionally(ex -> {
-                                                    future.completeExceptionally(ex);
-                                                    return null;
-                                                });
-                                    } else {
-                                        future.complete(metadata);
-                                    }
-                                });
-
-                                return future;
-                            });
-                });
+                .thenCompose(topicExists -> fetchPartitionedTopicMetadataAsync(topicName)
+                        .thenComposeAsync(metadata -> {
+                            // There are a couple of potentially blocking calls, which we cannot make from the
+                            // MetadataStore callback thread.
+                            // If topic is already exist, creating partitioned topic is not allowed.
+                            if (metadata.partitions == 0
+                                    && !topicExists
+                                    && !topicName.isPartitioned()
+                                    && pulsar.getBrokerService().isAllowAutoTopicCreation(topicName)
+                                    && pulsar.getBrokerService().isDefaultTopicTypePartitioned(topicName)) {
+                                return pulsar.getBrokerService().createDefaultPartitionedTopicAsync(topicName);
+                            } else {
+                                return CompletableFuture.completedFuture(metadata);
+                            }
+                        }, pulsar.getExecutor())
+                );
     }
 
     @SuppressWarnings("deprecation")
     private CompletableFuture<PartitionedTopicMetadata> createDefaultPartitionedTopicAsync(TopicName topicName) {
+        if (topicName.toString().contains(TopicName.PARTITIONED_TOPIC_SUFFIX)) {
+            return FutureUtil.failedFuture(new PulsarServerException.
+                    InvalidTopicNameException("Invalid topic name: " + topicName));
+        }
```

Comment:
The prompt message is not clear , like : `"Invalid topic name: " + topicName + ",  should not contain -partition-" `

Reply:
+1, We should give exact reason.

Reply:
fixed